src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useRef } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  setNewTodo: (value: string) => void;
+  newTodo: string;
+  setCurrentTodos: (value: React.SetStateAction<Todo[]>) => void;
+  setUpdatingTodo: (value: boolean) => void;
+  setErrorNotification: (value: string) => void;
+  updatingTodo: boolean;
+};
+
+export const Header: React.FC<Props> = ({
+  newTodo,
+  setNewTodo,
+  setCurrentTodos,
+  setUpdatingTodo,
+  setErrorNotification,
+  updatingTodo,
+}) => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputRef.current && !updatingTodo) {
+      inputRef.current.focus();
+    }
+  }, [updatingTodo]);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodo(e.target.value);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    if (newTodo.trim().length > 0) {
```

Comment:
I would recommend not checking the `valid` case, but instead check the `invalid` and if it is invalid you do what you need and then return

```
if (newTodo.trim().length === 0) {
  setErrorNotification('Title should not be empty');
  setTimeout(() => {
    setErrorNotification('');
  }, 3000);
  
  return;
}

The rest of the code which handle valid case
```